shm/cargo.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+  private: 
+     std::string name_;
+     size_t amount_;
```

Comment:
It is a wrong use of size_t. size_t is  for indexing arrays, vectors etc. https://en.cppreference.com/w/cpp/types/size_t
Ideally type of amount should be the measurement units.
so ideally: Kilograms amount_;   Then you can do: cargo1.amount_ + cargo2.amount_  But you cannot do cargo2.amount_ + cargo1.base_price_